### PR TITLE
2648 Enhance RandLambda to execute deterministic transforms for random part of dataset

### PR DESCRIPTION
**Pull request by Nic-Ma**
part of #2648 .

### Description
This PR enhanced the `RandLambdad` transform and also added array version `RandLambda` transform to execute transform for random part of dataset.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by rijobro** [comment]
Doesn't this imply that we should just implement `RandResize`, `RandCenterSpatialCropd` and `ThresholdIntensityd`?

**Comment by Nic-Ma** [comment]
Hi @rijobro ,

The basic idea of this `RandCompose` is to randomly execute transforms on the data, it's random prob "YES or NO", not like our other random size / random value transforms.
And with this `RandCompose`, no need to implement the "random" version transform for every deterministic transform.

Thanks.

**Comment by wyli** [comment]
my question is why is this implemented as a special compose method, in the example prob=0.2 for Resized, prob=0.3 for CenterSpatialCropd, and prob0.4 for ThresholdIntensityd are all independent events.  Instead, I think each of them could be implemented with a transform wrapper inheriting the `RandomizableTransform` interface https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/transform.py#L236 instead of

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your suggestion.
Actually, at the beginning, I planned to enhance the `RandLambdad` transform to support `prob`, but I feel maybe using `RandCompose` can help simplify a list of wrapper of transforms.
Anyway, if you guys prefer to wrapper, let me change back to enhance the `RandLambdad`.

Thanks.

**Comment by rijobro** [comment]
I think enhancing `RandLambda` would be more appropriate. Even though most users will combine transforms in a chain, we shouldn't assume that they will all of the time.

**Comment by Nic-Ma** [comment]
Hi @rijobro  and @wyli ,

I have updated this PR to enhance `RandLambdad` and also added the array version `RandLambda` transform.
Could you please help review it again?

Thanks in advance.

**Comment by wyli** [comment]
Thanks, would it be useful to make this invertible?

**Comment by wyli** [comment]
Was partly discussed here https://github.com/Project-MONAI/MONAI/discussions/2004

**Comment by Nic-Ma** [comment]
Hi @wyli @rijobro ,

I added inverse operation to the PR, could you please help review it again?

Thanks in advance.

File: monai/transforms/utility/dictionary.py [hunk 1]
```diff
@@ -852,51 +852,110 @@ class Lambdad(MapTransform):
...
+    Note: The inverse operation doesn't allow to define `extra_info` or access other information, such as the
+        image's original size. If need these complicated information, please write a new InvertibleTransform directly.
+
     """
 
     def __init__(
         self,
         keys: KeysCollection,
         func: Union[Sequence[Callable], Callable],
+        inv_func: Union[Sequence[Callable], Callable] = lambda x: x,
```
Comment (wyli) [inline monai/transforms/utility/dictionary.py:870]: the lambda might cause the issue https://github.com/Project-MONAI/MONAI/issues/2122

Comment (Nic-Ma) [reply monai/transforms/utility/dictionary.py]: Hi @wyli ,

Thanks for pointing this.
I updated in the latest commit.

Thanks.

**Review by wyli — APPROVED** [review]

**Pull request merged**